compiler/src/abi/mod.rs

```diff
@@ -0,0 +1,23 @@
+use crate::errors::CompileError;
+use vyper_parser as parser;
+
+mod json_builder;
+
+/// Builds a JSON ABI from the source file.
+///
+/// This API should be rethought as it only provides the ABI for the first contract in the
+/// source file. See: https://github.com/ethereum/rust-vyper/issues/12
+/// TODO: Improve the JSON ABI builder API.
+pub fn build(src: &str) -> Result<String, CompileError> {
```

Comment:
Hope I'm understanding this correctly, but I'm not sure if this is the right place for ethpm. In the solidity compiler - they are adopting the [ethpm v3 spec](http://ethpm.github.io/ethpm-spec/v3-package-spec.html) for their [`"metadata"` output](https://solidity.readthedocs.io/en/v0.6.2/metadata.html) rather than anything to do with the ABI structure, which is what it appears to me to be happening here. But, let me know if I'm misunderstanding or can help with any other questions.

Reply:
Nick, it might be interesting to start a "EthPM integration guide for compilers" to try and capture some of the ideas and concepts that might be useful for language designers to know but might not be immediately obvious.

The ideas that come to mind are:

- what the process would be for using an EthPM package as the input for compilation
- if the compiler generates an EthPM package, they should make sure they do....
    - what does compiler metadata look like and where should it be injected.
    - make sure you populate (sources, filesystem structure, contract types and their mappings back to their source files, etc?)
- suggested conventions for any areas where they may need to make a decision (such as any areas where unique identifiers need to be generated, etc).

Just a few passing thoughts.  Use them if they sound useful.